### PR TITLE
Adding the deployment shell script for Linux instances

### PR DESCRIPTION
**Pull request by mohanarpit**
This can definitely be simplified with fewer package installations based on user feedback

### CODE REVIEW

File: deploy/install.sh [hunk 1]
```diff
@@ -0,0 +1,183 @@
...
+echo ""
+read -p 'Would you like to setup a custom domain to access appsmith? [Y/n]: ' setup_domain
+setup_domain=${setup_domain:-Y}
+
+if [ $setup_domain == "Y" -o $setup_domain == "y" -o $setup_domain == "yes" -o $setup_domain == "Yes" ];then
+	read -p 'Enter your domain name (example.com): ' custom_domain
+fi
+
+NGINX_SSL_CMNT=""
+if [ ! $custom_domain ];then
```
Comment (sharat87) [inline deploy/install.sh:60]: Please revisit this condition. We are running the command inside `$custom_domain`, which I doubt is the intention here. We are probably checking whether the `$custom_domain` variable's value is empty.

```suggestion
if [[ -n $custom_domain ]]; then
```

File: deploy/install.sh [hunk 2]
```diff
@@ -0,0 +1,183 @@
...
+    fi
+
+done
+
+
+echo "++++++++++++++++++++"
+
+#echo "Running init-letsencrypt.sh...."
+cd $install_dir
+if [ $custom_domain ];then
```
Comment (sharat87) [inline deploy/install.sh:176]: Similar to my other comment in this script file, we probably want the below:

```suggestion
if [[ -z $custom_domain ]]; then
```

File: deploy/install.sh [hunk 3]
```diff
@@ -0,0 +1,183 @@
...
+fileInfo[/data/nginx/app.conf]="nginx_app.conf"
+fileInfo[/docker-compose.yml]="docker-compose.yml"
+fileInfo[/data/mongo/init.js]="mongo-init.js"
+fileInfo[/init-letsencrypt.sh]="init-letsencrypt.sh"
+fileInfo[/docker.env]="docker.env"
+
+for f in ${!fileInfo[@]}
+do
+
+    if [ -f $install_dir$f ]
```
Comment (sharat87) [inline deploy/install.sh:150]: I'd recommend inserting a `/` between `$install_dir` and `$f`. Linux/macOS don't choke on contiguous duplicate slashes in a file path, but a missing slash can be disastrous. We can optionally remove the slashes in the keys of the `fileInfo` definition.

```suggestion
    if [[ -f $install_dir/$f ]]
```

File: deploy/install.sh (hunk 1 above)
Comment (mohanarpit) [reply deploy/install.sh:92]: Done. Thanks for spotting this.

**Pull request merged**